### ISSUE TITLE
Using a lab folder instead of $home

### ISSUE THREAD

**Issue by arefg** [PowerShellOrg/PSKoans#183]
Since using any non-consistent elements such as these

"$Files = Get-ChildItem -Path $home -Recurse -Depth 2"

(Get-Command).Count

$x = Get-Command
$x[3].Name

in the code will not necessarily return the same value every time for everyone and every machine, and this will result in having to change things from time to time to be able to pass the previously completed tests which is really frustrating,  I suggest creating a temporary lab folder somewhere in $home, with some automatically populated files and folders to be used as a reference for all the PSKoans.

and for the Get-Command I can't think of any good solution right now but I think changing that will also help the overall integrity of the module.

**Comment by vexx32** [comment]
Yeah, that's something I've been thinking about for a while. Not entirely sure the best way to do it is, though; I don't _really_ want to clutter someone's filesystem overly much, though. It might be that to some extent simply pulling the user's koan directory files themselves might be most effective for a majority of tests? And for those that need to create files, maybe we could make a custom version of `New-TemporaryFile` and just create dummy files in a subdirectory of the koans folder or something.

**Comment by arefg** [comment]
That sounds like a great idea. :thumbsup:
